Framework/API/src/AlgorithmProxy.cpp

```diff
@@ -108,21 +109,31 @@ bool AlgorithmProxy::executeAsyncImpl(const Poco::Void &dummy) {
     throw;
   }
   stopped();
-  return m_isExecuted;
+  return (resultState() == ResultState::Success);
 }
 
+/// Gets the current execution state
+ExecutionState AlgorithmProxy::executionState() const {
+  return m_executionState;
+}
+
+/// Gets the current result State
+ResultState AlgorithmProxy::resultState() const { return m_resultState; }
+
 /// True if the algorithm is running.
 bool AlgorithmProxy::isRunning() const {
-  return m_alg ? m_alg->isRunning() : false;
+  return m_alg ? (m_alg->executionState() == ExecutionState::Running) : false;
```

Comment:
^ Esp as this returns if we are running

Reply:
Done as part of the review of the preceding PR, now merged into this one.

Reply:
And the whole class is being removed in the next PR that this is blocking